block_filtered_course_list.php

```diff
@@ -207,24 +207,21 @@ private function _process_filtered_list() {
         global $PAGE;
         $output = $PAGE->get_renderer('block_filtered_course_list');
 
-        // Parse the textarea settings into an array of arrays.
-        $filterconfigs = array_map(function($line) {
-            return array_map(function($item) {
-                return trim($item);
-            }, explode('|', $line, 2));
-        }, explode("\n", $this->fclconfig->filters));
-
+        // Parse the textarea settings into an array of lines.
+        $filterconfig = new \block_filtered_course_list\local\config_parser($this->fclconfig->filters);
```

Comment:
The change you are making here introduces a new class with three methods to accomplish much the same thing as the existing code. Please tell me what we are gaining by this change. We are not invoking this class anywhere but here and we are not reusing any of the methods. Are you thinking that we might eventually want to do that, or is there some other advantage I'm not thinking of?

Reply:
You're probably right. My goal was mainly to separate the logic into a single place. I'm sure a lib function or two would work just as well.